File: examples/file_adaptor.rb
#!/usr/bin/ruby -w
# Created by Luis Campo Giralte
# GPL License
# 
# Example for use DatabaseAdaptor for drop information 
require "../src/ruaiengine"

class OutputAdaptor < DatabaseAdaptor 
  attr_reader :total_inserts 
  attr_reader :total_updates 
  attr_reader :total_removes 

  def initialize
    @total_inserts = 0
    @total_updates = 0
    @total_removes = 0
  end

  def insert(flowid)
    printf "New Connection on %s\n", flowid
    @total_inserts += 1
  end
  def remove(flowid)
    printf "Remove Connection on %s\n", flowid
    @total_removes += 1
  end
  def update(flowid, data)
    printf "Update Connection on %s data:%s\n", flowid,data
    @total_updates += 1
  end
end

s = StackLan.new
pd = PacketDispatcher.new
pd.stack = s

s.total_tcp_flows = 32000
s.total_udp_flows = 32000

output = OutputAdaptor.new

s.set_tcp_database_adaptor(output)

pd.open("ens7")
begin
    pd.run()
rescue
    print "Stop capturing packets"
end

pd.close()

File: examples/file_adaptor.lua
#!/usr/bin/env lua

-- Example for use DatabaseAdaptor for drop information 

__author__ = "Luis Campo Giralte"
__copyright__ = "Copyright (C) 2013-2016 by Luis Campo Giralte"
__revision__ = "$Id$"
__version__ = "0.1"

luaiengine = require('luaiengine')

FileAdaptor = {}

FileAdaptor.new = function(self,filename)
    local self = {}
    setmetatable(self,luaiengine.DatabaseAdaptor)
    self.f = assert(io.open(filename, "w"))

    self.insert = function(key)
    end
    self.update = function(key,data)
        self.f:write(data.."\n")
    end
    self.remove = function(key)
    end
    return self
end


-- Load an instance of a Network Stack Lan 
st = luaiengine.StackLan()
pd = luaiengine.PacketDispatcher()

st.tcp_flows = 327680
st.udp_flows = 163840

pd:set_stack(st)

adap = FileAdaptor:new("tcpflows.dat")
st:set_udp_database_adaptor("adap")

pd:open("enp0s25") 
pd:run()
pd:close()

